docs/zero_knowledge.md

```diff
@@ -1,136 +1,76 @@
-
-When first described Zero-Knowledge proofs were interactive, meaning the 
-prover and verifier need to interact, like in our example above.
-Later it was discovered that a shared reference string is enough to achieve 
-computational zero-knowledge without requiring an interaction between prover 
-and verifier.
-
-Over time different Zero-Knowledge protocols have been discovered. They have 
-beautiful names like *STARK* (Scalable Transparent ARgument of Knowledge), 
-*BulletProof*, *SNARG* (Succinct Non-interactive ARGuments) and *SNARK* 
-(Succinct Non-interactive ARgument of Knowledge).  
-I will not explain the different protocols in detail here.
-Instead, I will just give a brief overview of what those fancy abbreviation 
-actually mean so you have a staring point for the next internet research 
-rabbit hole.
-
-- Scalable:  
-  Both prover and verrifier running times are scalable.
-- Transparent:  
-  Doesn't require a setup-phase which uses a non-public random parameter like 
-  the shared reference string mentioned above.
-- ARgument:  
-  Strictly mathematically speaking, many of the above protocols are no proofs 
-  but arguments. However in practice we can still think of them as a proofs.
-- Knowledge:  
-  The prover proofs that they themselves know the solution. In our above 
-  example that means that the prover doesn't prove that a solution exists 
-  but that they actually know the solution.
-- Succinct:  
-  The proofs are relatively small and easy to verify
-- Non-interactive:  
-  No interaction between prover and verifier is needed
+Conventional proof systems make sure that the prover cannot act maliciously. ZKP systems take this a step further, and also make sure the 
+verifier cannot act maliciously. After computing and sending a ZKP, the verifier (or an eavesdropper) only learns that the prover is in possession of $w$, nothing more. This fact has the following implications:
+
+- The verifier is not able to prove to a third party that he knows $w$ (because he does not know it). 
+- The verifier is not able to prove to a third party that the initial prover knows $w$.
+
+The second point might sound a bit strange at first, so let us look at it in more detail:
+
+Say you recorded the proving procedure of the coloring problem explained above, so you can later convince your very skeptical friends that I know a valid solution. On the recording they would see that every edge you chose indeed connects two vertices of different color. But it won't convince them (they are very skeptical) since you and I could have played false and agreed on the vertices to be uncovered beforehand.
+
+So, where do we want to go with it? Using ZKPs you would, for example, be able to prove that you know a password without ever giving it to any potentially malicious third party or eavesdropper. As such, you could prove your identity to just one person and he would not be able to prove that to anyone else, nor impersonate you.  
```

Comment:
not 'he' but 'they'. I know it sounds weird but I really want to write gender neutral and the more people do it, the more we will all get used to reading it